CMakeLists.txt

```diff
@@ -0,0 +1,451 @@
+set(CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/README.md")
+set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.md")
+set(CPACK_PACKAGE_VERSION_MAJOR "${ZIPPER_VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${ZIPPER_VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${ZIPPER_VERSION_PATCH}")
+set(CPACK_RPM_PACKAGE_LICENSE "MIT")
+set(CPACK_RPM_PACKAGE_GROUP "Libraries/Development")
+set(CPACK_DEBIAN_PACKAGE_SECTION "Libraries")
+
+set(CPACK_SOURCE_IGNORE_FILES "${CMAKE_CURRENT_BINARY_DIR};/.svn/;/.libs/;/.deps/;/.bzr/;.*.o$;.*.lo$;.*.la$;/.git/;${CPACK_SOURCE_IGNORE_FILES};/.DS_Store;/.svnignore;blib;libsbml-dist;*.txt.user")
+
+set(ADDITIONAL_LIB_DIRS)
+if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+    set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "i386")
+    set(CPACK_RPM_PACKAGE_ARCHITECTURE "i386")
+else()
+    set(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "amd64")
+    set(CPACK_RPM_PACKAGE_ARCHITECTURE "x86_64")
+    set(ADDITIONAL_LIB_DIRS "/usr/lib64" "/usr/lib/x86_64-linux-gnu/")
+endif()
+
+include(CPack)
+
+# Add an option to compile with all warnings shown
+option(WITH_WALL     "Compile with -Wall, so that the compiler will display all warnings." OFF)
+mark_as_advanced(WITH_WALL)
+
+if(WITH_WALL)
+    if(MSVC OR USING_INTEL)
+        add_definitions(/W4)
+    else()
+        add_definitions(-Wall)
+    endif()
+endif()
+
+if (UNIX)
+  set(ZIPPER_LIBRARY Zipper)
+else()
+  set(ZIPPER_LIBRARY libZipper)
+endif()
+# Set build type default.
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+  "Choose the type of build to perform. The options are: None (CMAKE_CXX_FLAGS
+or CMAKE_C_FLAGS are used), Debug, Release, RelWithDebInfo, MinSizeRel.")
+if("${CMAKE_BUILD_TYPE}" STREQUAL "")
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+    "Choose the type of build to perform. The options are: None (CMAKE_CXX_FLAGS
+or CMAKE_C_FLAGS are used), Debug, Release, RelWithDebInfo, MinSizeRel." FORCE)
+endif()
+
+
+find_library(LIBZ_LIBRARY
+    NAMES zdll.lib z zlib.lib
+    PATHS /usr/lib /usr/local/lib
+          ${CMAKE_OSX_SYSROOT}/usr/lib
+          ${ZIPPER_DEPENDENCY_DIR}/lib
+          ${ADDITIONAL_LIB_DIRS}
+          $ENV{ZLIBROOT}/lib
+    DOC "The file name of the zip compression library."
+)
+
+
+if (NOT LIBZ_INCLUDE_DIR)
+  find_path(LIBZ_INCLUDE_DIR
+      NAMES zlib.h zlib/zlib.h
+      PATHS ${CMAKE_OSX_SYSROOT}/usr/include
+            /usr/include /usr/local/include
+            ${ZIPPER_DEPENDENCY_DIR}/include
+            $ENV{ZLIBROOT}/include
+            NO_DEFAULT_PATH
+      DOC "The directory containing the zlib include files."
+            )
+endif()
+    
+if (NOT LIBZ_INCLUDE_DIR)
+    find_path(LIBZ_INCLUDE_DIR
+    NAMES zlib.h zlib/zlib.h
+    PATHS ${CMAKE_OSX_SYSROOT}/usr/include
+          /usr/include /usr/local/include
+          ${ZIPPER_DEPENDENCY_DIR}/include
+          $ENV{ZLIBROOT}/include
+    DOC "The directory containing the zlib include files."
+          )
+endif()
```

Comment:
The difference is in the use of the NO_DEFAULT_PATH, basically I want to make sure that I really take the one that I have provided rather than another one that is found. That was needed in another project I'm involved with. more information below. 

1) https://cmake.org/cmake/help/v3.0/command/find_path.html